xarray/tests/test_dataset.py

```diff
@@ -3261,3 +3264,132 @@ def test_dir_unicode(data_set):
+    actual = getattr(rolling_obj, name)()
+    if key is 'z1':  # z1 does not depend on 'Time' axis. Stored as it is.
+        expected = ds[key]
+    elif key is 'z2':
+        expected = getattr(bn, func_name)(ds[key].values, window=7, axis=0,
+                                          min_count=min_periods)
+    assert_array_equal(actual[key].values, expected)
+
+    # Test center
+    rolling_obj = ds.rolling(time=7, center=center)
+    actual = getattr(rolling_obj, name)()['time']
+    assert_equal(actual, ds['time'])
+
+
+def test_rolling_invalid_args(ds):
+    pytest.importorskip('bottleneck')
+    with pytest.raises(ValueError) as exception:
+        ds.rolling(time=7, min_periods=1).median()
+    assert 'Rolling.median does not' in str(exception)
+
+
+@pytest.mark.parametrize('center', (True, False))
+@pytest.mark.parametrize('min_periods', (None, 1, 2, 3))
+@pytest.mark.parametrize('window', (1, 2, 3, 4))
+def test_rolling_pandas_compat(center, window, min_periods):
+    df = pd.DataFrame({'x': np.random.randn(20), 'y': np.random.randn(20),
+                       'time': np.linspace(0, 1, 20)})
+    ds = Dataset.from_dataframe(df)
+
+    if min_periods is not None and window < min_periods:
+        min_periods = window
+
+    df_rolling = df.rolling(window, center=center,
+                            min_periods=min_periods).mean()
+    ds_rolling = ds.rolling(index=window, center=center,
+                            min_periods=min_periods).mean()
+    # pandas does some fancy stuff in the last position,
+    # we're not going to do that yet!
+    np.testing.assert_allclose(df_rolling['x'].values[:-1],
+                               ds_rolling['x'].values[:-1])
+    np.testing.assert_allclose(df_rolling.index,
+                               ds_rolling['index'])
+
+
+@pytest.mark.parametrize('ds', (1, 2), indirect=True)
+@pytest.mark.parametrize('center', (True, False))
+@pytest.mark.parametrize('min_periods', (None, 1, 2, 3))
+@pytest.mark.parametrize('window', (1, 2, 3, 4))
+@pytest.mark.parametrize('name',
+                         ('sum', 'mean', 'std', 'min', 'max', 'median'))
+def test_rolling_reduce(ds, center, min_periods, window, name):
+
+    if min_periods is not None and window < min_periods:
+        min_periods = window
+
+    # std with window == 1 seems unstable in bottleneck
+    if name == 'std' and window == 1:
+        window = 2
+    if name == 'median':
+        min_periods = None
+
+    rolling_obj = ds.rolling(time=window, center=center,
+                             min_periods=min_periods)
+
+    # add nan prefix to numpy methods to get similar behavior as bottleneck
+    actual = rolling_obj.reduce(getattr(np, 'nan%s' % name))
+    expected = getattr(rolling_obj, name)()
+    assert_allclose(actual, expected)
+    assert ds.dims == actual.dims
+    # make sure the order of data_var are not changed.
+    assert list(ds.data_vars.keys()) == list(actual.data_vars.keys())
```

Comment:
Here is the test to make sure that the order of `data_vars` is not changed by `rolling().reduce()`.